ambry-api/src/main/java/com.github.ambry/config/ClusterMapConfig.java

```diff
@@ -66,11 +63,72 @@
   @Default("")
   public final String clusterMapSslEnabledDatacenters;
 
+  /**
+   * The clustermap participant factory to use if registering as a participant.
+   */
+  @Config("clustermap.participant.factory")
+  @Default("com.github.ambry.clustermap.StaticParticipantFactory")
+  public final String clusterMapParticipantFactory;
+
+  /**
+   * The clustermap clustermanager factory to use for instantiating the Cluster Manager.
+   */
+  @Config("clustermap.clustermanager.factory")
+  @Default("com.github.ambry.clustermap.StaticClusterManager")
+  public final String clusterMapClusterManagerFactory;
+
+  /**
+   * Serialized json containing the information about all the zk hosts that the Helix based cluster manager should
+   * be aware of. This information should be of the following form:
+   *
+   * {
+   *   "zkInfo" : [
+   *     {
+   *       "datacenter":"dc1",
+   *       "zkConnectStr":"abc.example.com:2199",
+   *     },
+   *     {
+   *       "datacenter":"dc2",
+   *       "zkConnectStr":"def.example.com:2300",
+   *     }
+   *   ]
+   * }
+   *
+   */
+  @Config("clustermap.dcs.zk.connect.strings")
+  @Default("")
+  public final String clusterMapDcsZkConnectStrings;
+
+  /**
+   * The name of the associated cluster for this node.
+   */
+  @Config("clustermap.cluster.name")
+  public final String clusterMapClusterName;
+
+  /**
+   * The name of the associated datacenter for this node.
+   */
+  @Config("clustermap.datacenter.name")
```

Comment:
this looks like repeated info since the router has the same info. One of them should no longer have the config (preferably, the router doesn't need to). Otherwise, there is a risk of diverging configs?

Same thing with host name and port.